searchlib/src/main/java/com/yahoo/searchlib/aggregation/Group.java

```diff
@@ -261,18 +282,20 @@ public Group addOrderBy(ExpressionNode exp, boolean asc) {
         if (exp instanceof AggregationResult) {
             exp = new AggregationRefNode((AggregationResult)exp);
         }
-        exp.select(REF_LOCATOR, new RefResolver(this));
-        orderByExp.add(exp);
-        orderByIdx.add((asc ? 1 : -1) * orderByExp.size());
+        RefResolver refResolver = new RefResolver(aggregationResults);
+        exp.select(REF_LOCATOR, refResolver);
+        aggregationResults = refResolver.results;
+        orderByExp = add(orderByExp, exp);
+        orderByIdx = add(orderByIdx, (asc ? 1 : -1) * orderByExp.size());
         return this;
     }
 
     public List<Integer> getOrderByIndexes() {
-        return Collections.unmodifiableList(orderByIdx);
+        return List.copyOf(orderByIdx);
```

Comment:
Isn't it better to keep Collections-unmodifiableList here? This will make a copy each time the code checks whether this list is non-empty etc.

Reply:
For the `List.of()` variants, this doesn't actually copy, but all large lists will be copied, indeed.

Reply:
Not if the list is already unmodifiable, which is the most likely case. Then it will only return the argument.

Reply:
If it is already unmodifiable both ways are cheap, otherwise List.of is not.